### PR TITLE
address review feedback (part 2)

### PR DESCRIPTION
**Pull request by aramase**
Note from the sync on pending comments:

- Add https
    - Imagine hosting well known and jwks endpoint locally in the cluster
    - Service oidc and ns oidc
- Distributed claims; type resolver
- Add the CEL examples for groups
- Remove TODO and mention username empty is invalid
    - Today empty username invalid?
- No need to make iat and nbf required unless already required
- Add metrics for hot reload
- Check the rollback scenarios
- How can a rollout or rollback fail? Can it impact already running workloads?
- Add https://github.com/kubernetes/enhancements/pull/3332#discussion_r1082106308 to the KEP
- Add notes for CEL
- Add note about upper bound of 5s for CEL eval
- Handle this https://github.com/kubernetes/enhancements/pull/3332#discussion_r1082124562 by defaulting maybe? Add a TODO
- Document which network proxy is going to be used

### CODE REVIEW

**Comment by aramase** [comment]
cc @enj

File: keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md [hunk 1]
```diff
@@ -340,9 +335,19 @@ type JWTAuthenticator struct {
...
         // discoveryURL if specified, overrides the URL used to fetch discovery information.
+        // This is in scenarios where the well-known and jwks endpoints are hosted locally in the cluster.
         // Format must be https://url/path.
+        //
+        // The connections from the API server to service will run over a secure HTTPS connection
+        // by prefixing https: to the service name in the API URL, but they will not validate the
+        // certificate provided by the HTTPS endpoint nor provide client credentials. So while the
+        // connection will be encrypted, it will not provide any guarantees of integrity.
+        // These connections are not currently safe to run over untrusted or public networks.
+        // xref: https://kubernetes.io/docs/concepts/architecture/control-plane-node-communication/#api-server-to-nodes-pods-and-services
```
Comment (enj) [inline keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md]: Uh... we would validate the TLS connection as normal - the hostname on the leaf cert would need to match `oidc.oidc-namespace` in this example and be verified by the given CA data.

File: keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md [hunk 2]
```diff
@@ -432,6 +437,21 @@ type JWTAuthenticator struct {
         // Expression must produce a string or string array value.
         // "", [], missing, and null values are treated as having no groups.
         // TODO: investigate if you could make a single expression to construct groups from multiple claims. If not, maybe []PrefixedClaimOrExpression?
+        // For input claim:
+        // {
+        //     "claims": {
+        //     "roles":"foo,bar",
+        //     "other_roles":"baz,qux"
+        //     "is_admin": true
+        //     }
```
Comment (enj) [inline keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md]:
```suggestion
        //     "claims": {
        //         "roles":"foo,bar",
        //         "other_roles":"baz,qux"
        //         "is_admin": true
        //     }
```

File: keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md [hunk 3]
```diff
@@ -340,9 +335,19 @@ type JWTAuthenticator struct {
         URL string `json:"url,omitempty"`
```
Comment (enj) [inline keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md]: Document which network proxy is going to be used?

File: keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md [hunk 4]
```diff
@@ -552,14 +592,26 @@ type JWTAuthenticator struct {
...
+  * Using `has()` to guard access to fields.
+  * Could we do some kind of defaulting for fields that don't exist?
+
+> Notes from PR review (jpbetz):
+>
+> You can pass a context to CEL and cancel runtime evaluation if the context is canceled. This causes the CEL expression to halt execution promptly and evaluate to an error.
+> You can also put a runtime limit (measured in abstract cost units that are hardware and wall clock independent) on CEL expressions to bound running time.
+> (There is also a way to set a limit for the estimated cost, which is computed statically on compiled CEL programs if you know the worst case size of the input data, but this might be overkill for this feature)
 
 ### Flags
```
Comment (enj) [inline keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md:618]: For rollback, mention that the if they were using the pre-existing OIDC flags before, that they would need to add those back.

**Review by enj — CHANGES_REQUESTED** [review]

**Pull request merged**